lib/ClangImporter/ImportDecl.cpp

```diff
@@ -322,8 +322,11 @@ getSwiftStdlibType(const clang::TypedefNameDecl *D,
     break;
 
   case MappedCTypeKind::VaList:
-    if (ClangTypeSize != ClangCtx.getTypeSize(ClangCtx.VoidPtrTy))
-      return std::make_pair(Type(), "");
+   if (ClangTypeSize != ClangCtx.getTypeSize(ClangCtx.VoidPtrTy)) {
```

Comment:
Indentation here is off.

We shouldn't be doing a size-based validation here; we should be asking whether this type is equal to `ClangCtx.getBuiltinVaListType()`.  If there are platforms where that doesn't work (because of slight pointer-size differences?) we should call them out specially instead of doing more work here.

Reply:
Doesn't my approach effectively accomplish that?  Seems like both of us have the same underlying motivation though.

Reply:
Well, but your approach still hardcodes target differences that we don't really need to care about in the importer.  All we really care about for portability is that the stdlib actually implements `CVaListPointer` correctly for all the different kinds of `va_list`, and that's easier to metaprogram directly based on Clang's existing knowledge of the target's `va_list` kind than it is to write a bunch of awkward platform conditions that essentially duplicate it.